### PR TITLE
make errors in phiSpy non-fatal. Annotation can continue afterwards

### PR DESCRIPTION
**Pull request by landml**
@scanon, @sychan -  Please merge. An error in phiSpy is causing the entire RAST annotation to fail without returning output from the other tools.

### CODE REVIEW

**Comment by landml** [comment]
Added code so none of the error coming back from a single tool cause the entire annotation to fail. This can be a problem when annotating a long list of genomes with the same tools. Failures of an individual genome in the middle can be easily overlooked.

**Comment by JamesJeffryes** [comment]
I think it's confusing to have a failure_not_fatal flag if it will always be ignored I recommend either adding it to all the calls or removing it entirely

**Comment by landml** [comment]
@JamesJeffryes  - This is an artifact of the order in which I found the errors. Some of the tools that already have the change I made for phispy (failure_is_not_fatal = true) are still failing and causing problems for RAST. I didn't want to make massive surgery on the revered kb_seed/genome_annotation until 1) I saw if this really is the source of the problem and 2) I better understood the nearly 30,000 lines of code. I'm trying to work in small reversible steps. I **could** take out all the references to failure_is_not_fatal  but how long would it take to find the unintended consequences?

The commit to phispy is really not needed if the second commit is merged. It merely makes the phispy tool look like the other tools in that section of code.

**Comment by JamesJeffryes** [comment]
I'm just take the perspective of a future dev trying to figure out why failure_is_not_fatal exists. I'd don't have merge powez so we'll have to apeal to a higher power anyway

**Comment by landml** [comment]
If it fixes the problems, I can clear up the confusion next.

**Comment by landml** [comment]
@JamesJeffryes, @scanon , @sychan - Can I get this change merged so I can close a ticket and a weakness of the RAST annotation pipeline? We have a refactor of RAST scheduled for the tech debt this quarter and I can spend more time then finding all the potential pitfalls in clearing up what looks like the possible confusion mentioned above.

File: lib/Bio/KBase/GenomeAnnotation/GenomeAnnotationImpl.pm [hunk 1]
```diff
@@ -26114,18 +26114,19 @@ sub run_pipeline
 
 	    if ($@)
 	    {
-		if ($stage->{failure_is_not_fatal})
-		{
-		    warn "Error invoking method $method: $@\nContinuing because failure_is_not_fatal flag is set";
+#		if ($stage->{failure_is_not_fatal})
+#		{
+#		    warn "Error invoking method $method: $@\nContinuing because failure_is_not_fatal flag is set";
```
Comment (ialarmedalien) [inline lib/Bio/KBase/GenomeAnnotation/GenomeAnnotationImpl.pm:26119]: It's better to remove code completely than comment it out -- git will contain the previous versions of the code if people need to look back at it, and having a load of commented-out lines increases cognitive load on those looking through the codebase. Ideally code should be easy to read and understandable to all.

**Review by ialarmedalien — CHANGES_REQUESTED** [review]
Remove lines instead of commenting them out to help maintain a clean codebase.

Thanks! :)

**Comment by landml** [comment]
@ialarmedalien - Commented lines removed.

**Pull request closed without merging**